agent/agent.go

```diff
@@ -518,6 +518,25 @@ func (a *Agent) runPostTaskCommands(ctx context.Context, tc *taskContext) {
 			"total_time": time.Since(start).String(),
 		})
 	}
+
+	a.killProcs(postCtx, tc, false)
```

Comment:
This is already in a defer. It might be worth refactoring the below into its own function so we can still defer but not call this twice

Reply:
I made it kill the processes here, because if for some reason the post task is modifying the task directory, it can interfere with the task sync.